packages/deployment/upgrade-test/Dockerfile

```diff
@@ -9,96 +13,111 @@ RUN echo "${BOOTSTRAP_MODE}"
 RUN mkdir -p /usr/src/agoric-sdk/upgrade-test-scripts
 WORKDIR /usr/src/agoric-sdk/
 COPY ./start_ag0.sh ./upgrade-test-scripts/
-COPY ./bash_entrypoint.sh ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
+COPY ./bash_entrypoint.sh ./env_setup.sh ./start_to_to.sh ./package.json ./*.mjs ./upgrade-test-scripts/
 SHELL ["/bin/bash", "-c"]
+# this is the only layer that starts ag0
 RUN . ./upgrade-test-scripts/start_ag0.sh
 
+## UPGRADE LAYER
 ## this is agoric-upgrade-8 aka pismoA
 FROM ghcr.io/agoric/agoric-sdk:29 as agoric-upgrade-8
 ARG BOOTSTRAP_MODE
+# ??? why doesn't this set UPGRADE_TO ?
 ENV THIS_NAME=agoric-upgrade-8 BOOTSTRAP_MODE=${BOOTSTRAP_MODE}
-WORKDIR /usr/src/agoric-sdk/
+# copy from previous build
+COPY --from=agoric-upgrade-7-2 /root/.agoric /root/.agoric
 
-COPY ./bash_entrypoint.sh ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
+#region boilerplate to copy scripts for this upgrade
+WORKDIR /usr/src/agoric-sdk/
+COPY ./bash_entrypoint.sh ./env_setup.sh ./start_to_to.sh ./package.json ./*.mjs ./upgrade-test-scripts/
 COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-COPY --from=agoric-upgrade-7-2 /root/.agoric /root/.agoric
```

Comment:
AFAIK, this very much needs to be copied as it's the state of ag0/agd

Reply:
> this very much needs to be copied
It is copied on green line 28 (just a moved line, not a deletion), but this threw me for a loop as well.

It looks like @turadg was trying to reorder each of these sections so that they matched one another.